src/middleware.ts

```diff
@@ -0,0 +1,5 @@
+export { default } from 'next-auth/middleware'
+
+export const config = {
+  matcher: ['/admin', '/profile'],
+}
```

Comment:
@dimitur2204 Here's a suggestion for the auth middleware to support the previous functionality - the `isAdmin` check

```suggestion
import { getToken } from 'next-auth/jwt'
import withAuth from 'next-auth/middleware'

export default withAuth({
  pages,
  callbacks: {
    authorized: async ({ req }) => {
        const session = await getToken({ req })
        if (!session) return false
        if (session.accessTokenExpires <= Date.now()) return false

        if (req.nextUrl.pathname.startsWith('/profile')) {
          return true
        }

        if (req.nextUrl.pathname.startsWith('/admin')) {
          return isAdmin(session)
        }
    },
  },
})

export const config = {
  matcher: ['/admin', '/admin/:path*', '/profile', '/profile/:path*'],
}
```

Reply:
I will test that!

Reply:
Good catch it does work! Changed in [1123 - update auth to now protect admin](https://github.com/podkrepi-bg/frontend/pull/1125/commits/7a564396c4f3246008424e766d3a695c526b74ac)